### ISSUE TITLE
Make it easier to create new Pages directly from Navigation block

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#66923]
For many users the website menu is a key element on their site. We should make it easier for users to quickly scaffold out their site Pages directly from the Navigation without having to leave the block.

## Problem

However, currently we have a disconnect between the Menu and Pages - to add a new Page to your menu you must either:

- Go to Pages -> Create New -> Publish -> Return to Nav block -> Add Link -> Find new Page -> Add to Menu
- Add a new Menu Item -> Type name of page you want --> Click "Create draft" -> Go to Pages -> Find your Page -> Publish you Page 

Both process are incredibly convoluted and laborious. 

<details>
<summary>Video demonstrating process of adding a new Page from Navigation block</summary>

https://github.com/user-attachments/assets/7b6b3b20-545a-48c8-bc7e-e63e5845bd23
</details>

## Proposal

It should be possible to have more control when adding new Pages directly from the Navigation. To do this I propose we introduce a flow similar to that we currently have for adding `Blocks` in the Navigation block:

I propose that we remove the current "Draft" item from the search suggestions - it's confusing for users and - as a result - under-utilised. 

We replace this with the same kind of UI we have for `Add Block` but the text would read "Create new Page". 

![Image](https://github.com/user-attachments/assets/dbdaedf8-3441-432b-88b4-05fcc5880a64)

When clicked you would be shown a form (see example image below) allowing you to enter a title for your new Page _as well as_ an option allowing you to decide whether or not the page is "published".

![Image](https://github.com/user-attachments/assets/a43d55d3-11d0-4e07-a155-a88f7ddfcb24)

In addition, I also propose that we devise a less intrusive method of marking a Navigation item as being "Draft". For example, "incomplete" Social Icons are simply "greyed out". I think we could use a _complementary_ visual affordance to help users understand when a Page is "draft" without having to use the full suffix "Draft" which feels really clunky.

![Image](https://github.com/user-attachments/assets/57e2a61c-c324-4e7a-8473-baa38d25a764)

## Benefits

- Removes clutter and confusion caused by "Draft new Page" in the current search results
- Makes the process of Adding a new Page clearer and a more definite action.
- Allows users to avoid the overhead involved in publishing the draft Pages by simply marking them as "Published" when creating them.
- Enables quick scaffolding of a Navigation Menu and site Pages.

**Comment by getdave** [comment]
cc @richtabor who I seem to remember having a mock up for "Create page" and may have additional insight into this.

**Comment by getdave** [comment]
This screencapture from @jasmussen's [Figma file](https://www.figma.com/proto/46qRP607dtqHGEosEaWWhY/Product-Design?page-id=1213%3A1072&type=design&node-id=1215-21009&viewport=228%2C-561%2C0.5&t=re4JzAFqNwZdcNtd-1&scaling=min-zoom&starting-point-node-id=1215%3A21009) also shows how this idea fits into the wider Nav block UX

https://github.com/user-attachments/assets/7d487b16-f141-4214-aa15-446e8e7d25b8

**Comment by porg** [comment]
> scaffold out their site Pages directly from the Navigation

... and body inline links too... if I may add.

Or in other words: WordPress coming from a blog software and its "post-centric" paradigms which later added "page-centric" workflows too but only half heartedly finally allowing what wikis always allowed: Create your website structure on the fly, simply by creating links to pages which yet don't exist. That would be so desired.

To properly create pages from inline links on the fly please consider:
- #46731

**Comment by porg** [comment]
> I think we could use a complementary visual affordance to help users understand when a Page is "draft" without having to use the full suffix "Draft" which feels really clunky.

Exactly that user need I had 1-2 years ago when I started building my WordPress based website and I even achieved that with a workaround:

On my staging server I denoted those pages which in reality are drafts but in order to appear in the menu need to be published with a bullet "•" or asterisk "*" as a suffix. Small enough to realistically simulate my menu layouts. But still noticable enough to indicate the "incomplete/placeholder" nature of them. And then unpublished all those with the suffix in their title (easy for batch find and select purposes) before pushing to the production server.

Offtopic but worth mentioning here: I also wish one could work with drafts and show them normally in the menu for selected roles and search them to realistically simulate one's website "as it shall be when editorial is complete", and then gradually publish pages when they are ready. 

Currently this workflow is totally unsupported. And only possible with plugins and a staging/production setup. But not supported natively sadly. Natively WordPress basically still only supports a "live immediately" workflow (blog software origins) and every other workflow is totally fragmented. Drafts are only accessible in the backend.  That feels very early 2000ies. I have a detailed description of the use case. Interested? Honestly that would also be a project for which I as a UX designer would like to join you if you say you have a desire to develop further into this direction. But also if not I still would share my thoughts on this with you.

**Comment by getdave** [comment]
> Drafts are only accessible in the backend. That feels very early 2000ies. I have a detailed description of the use case. Interested? Honestly that would also be a project for which I as a UX designer would like to join you if you say you have a desire to develop further into this direction. But also if not I still would share my thoughts on this with you.

@porg Yes thank. It would be great to see this as a separate Issue so as to keep this one focused 👍
